raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -2033,6 +2033,47 @@ public void testUnattachedAsVoterCanBecomeFollowerAfterFindingLeader(boolean wit
         assertTrue(context.client.quorum().isFollower());
     }
 
+    @ParameterizedTest
+    @ValueSource(booleans = { true, false })
+    public void testUnattachedWithLeaderCanBecomeFollowerAfterFindingLeader(boolean withKip853Rpc) throws Exception {
+        int localId = randomReplicaId();
+        int otherNodeId = localId + 1;
+        int leaderNodeId = localId + 2;
+        int epoch = 5;
+        Set<Integer> voters = Set.of(localId, otherNodeId, leaderNodeId);
+        List<InetSocketAddress> bootstrapServers = voters
+            .stream()
+            .map(RaftClientTestContext::mockAddress)
+            .toList();
+
+        RaftClientTestContext context = new RaftClientTestContext.Builder(localId, voters)
+            .withBootstrapServers(Optional.of(bootstrapServers))
+            .withElectedLeader(epoch, leaderNodeId)
+            .withKip853Rpc(withKip853Rpc)
+            .build();
+
+        // after fetch timeout, node will become prospective with leader
+        context.time.sleep(context.fetchTimeoutMs);
+        context.client.poll();
+        assertTrue(context.client.quorum().isProspective());
+        assertEquals(leaderNodeId, context.client.quorum().leaderId().getAsInt());
+        assertEquals(epoch, context.currentEpoch());
+
+        // after election loss node will become follower because it had a last known leader
+        context.time.sleep(context.electionTimeoutMs() * 2L);
+        context.client.poll();
+        assertTrue(context.client.quorum().isFollower());
+        assertEquals(leaderNodeId, context.client.quorum().leaderId().getAsInt());
+        assertEquals(epoch, context.currentEpoch());
+
+        // node will send fetch request to leader
+        context.pollUntilRequest();
```

Comment:
Why not `pollUntilRequest` and check that the destination is the leader and the epoch is `epoch`?

We should try to avoid testing internal KRaft state in protocol suites like `KafkaRaftClientTest`.